addons/mail/models/mail_mail.py

```diff
@@ -473,6 +473,18 @@ def _prepare_outgoing_list(self, mail_server=False, recipients_follower_status=N
                 'partner_id': partner,
             })
 
+        # update headers with information coming from recipients, used to tweak
+        # SMTP versus Msg['To']. Consider all outgoing emails for this mail.mail.
+        if headers.get('X-Msg-To-Consolidate'):
```

Comment:
it will crash if the headers `X-Msg-To-Add` is not there
(we should check in the condition and do not consolidate if we don't give `X-Msg-To-Add`, or do not try do add `X-Msg-To-Add` if it's not there)